packages/node-core/src/meta/mmrQuery.service.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import assert from 'assert';
+import {Controller} from '@nestjs/common';
+import {u8aToHex} from '@subql/utils';
+import {MMR} from '@subql/x-merkle-mountain-range';
+import {Sequelize} from '@subql/x-sequelize';
+import {MmrStoreType, NodeConfig} from '../configure/NodeConfig';
+import {MmrProof, MmrPayload} from '../events';
+import {baseMmrService, keccak256Hash, PgBasedMMRDB} from '../indexer';
+import {getExistingProjectSchema} from '../utils';
+
+// Currently only serve MMR query
+export class PgMmrQueryDb extends PgBasedMMRDB {
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async setLeafLength(length: number): Promise<number> {
+    throw new Error('setLeafLength in PgMmrQueryDb should never been called');
+  }
+}
+
+@Controller('mmrs')
+export class MmrQueryService extends baseMmrService {
+  constructor(nodeConfig: NodeConfig, private sequelize: Sequelize) {
+    super(nodeConfig);
+  }
+
+  async init(offset: number): Promise<void> {
+    this._blockOffset = offset;
+    await this.ensureMmr();
+  }
+
+  async ensureMmr(): Promise<void> {
+    if (this._mmrDb) {
+      return;
+    }
+    this._mmrDb =
+      this.nodeConfig.mmrStoreType === MmrStoreType.Postgres
+        ? await this.ensurePostgresQueryMmr()
+        : await this.ensureFileBasedMmr(this.nodeConfig.mmrPath);
+  }
+
+  private async ensurePostgresQueryMmr(): Promise<MMR<PgMmrQueryDb>> {
+    const schema = await getExistingProjectSchema(this.nodeConfig, this.sequelize);
+    assert(schema, 'Unable to check for MMR table, schema is undefined');
```

Comment:
Lets be clearer about error messages. If a user got this error, they wouldn't know what to do. 
```suggestion
    assert(schema, 'Unable to find postges based MMR table. Do you need to migrate from file based MMR?');
```